test/suites/smoke/test-polkadot-decoding.ts

```diff
@@ -43,22 +107,68 @@ describeSuite({
             const keys = Object.keys(module[fn]);
             try {
               if (keys.includes("keysPaged")) {
-                const startKey = "";
+                // Generate a first query with an empty startKey
+                currentStartKey = "";
+                const emptyKeyEntries = await module[fn].entriesPaged({
+                  args: [],
+                  pageSize,
```

Comment:
This way we are decoding a total number of 1000 keys for each pallet storage point (sending 2 requests), but it is probably fine.

Reply:
Yes, but not in all cases, only in the bigger ones. I would say half of those which support `keysPaged` get fully checked on the first request and the randomized one is skipped.